Add classifiers as package data

sjspielman opened the pull request:
Closes #3 
Stacked on #8 

This PR adds a script in the `data-raw` directory to prepare the classifier `Rda` data files for use in the package. The files themselves, which I made very sure are _only_ classifiers and do not contain sensitive data!, are in `data`.
Note that the associated documentation will be added in #10.

In this code, I name the classifier objects `classifier.ktsp_unweighted` and `classifier.rf_weighted`. Users will have access to them as exported objects, but they shouldn't _need_ to access them directly but will just use the classifier functions.

sjspielman committed:
add data-raw to rbuildignore

sjspielman committed:
description updated

sjspielman committed:
add script to grab the classifiers

sjspielman committed:
add the RDA files

sjspielman committed:
renv environment

sjspielman committed:
Merge main, address conflict

sjspielman committed:
need one more update to the license field

allyhawkins commented on data-raw/classifiers.R:
```diff
@@ -0,0 +1,37 @@
+# Download file
+temp_file <- tempfile()
+s3_path <- "s3://data-lab-mb-ssp/models"
+sync_call <- glue::glue("aws s3 sync '{s3_path}' '{temp_file}' --exclude '*' --include 'baseline.rds' --exact-timestamps --profile {opt$aws_profile}")
```
Really minor, but I might make this part optional and only add the `--profile` flag if one is specified. I'm not sure how `aws s3 sync` will respond to an empty profile.

sjspielman replied:
It responds thusly:

```
usage: aws [--profile PROFILE] [--debug]

aws: error: argument --profile: expected one argument
```

I think I favor updating the script to require make `profile` required instead, so I'll do that. It's worth bearing in mind this script is only run _once_ ever (when making this PR in fact!), as only 3 people in the world can even run it at all 😬 .

allyhawkins commented on data-raw/classifiers.R:
```diff
@@ -0,0 +1,37 @@
+opt <- parse_args(OptionParser(option_list = option_list))
+
+# Download file
+temp_file <- tempfile()
```
It looks like you are using this as a temp directory to copy over the `baseline.rds` file, so I might be more specific here and use `tempdir()`. 
```suggestion
temp_dir <- tempdir()
```

allyhawkins commented on data-raw/classifiers.R:
```diff
@@ -0,0 +1,37 @@
+# Download file
+temp_file <- tempfile()
+s3_path <- "s3://data-lab-mb-ssp/models"
+sync_call <- glue::glue("aws s3 sync '{s3_path}' '{temp_file}' --exclude '*' --include 'baseline.rds' --exact-timestamps --profile {opt$aws_profile}")
```
```suggestion
sync_call <- glue::glue("aws s3 sync '{s3_path}' '{temp_dir}' --exclude '*' --include 'baseline.rds' --exact-timestamps --profile {opt$aws_profile}")
```
If you take my previous suggestion.

allyhawkins commented on data-raw/classifiers.R:
```diff
@@ -0,0 +1,37 @@
+system(sync_call)
+
+# Extract baseline model
+baseline_models <- readRDS(file.path(temp_file, "baseline.rds"))
```
```suggestion
baseline_models <- readRDS(file.path(temp_dir, "baseline.rds"))
```

allyhawkins reviewed (commented):
This looks mostly good, I just had a few small comments that we might want to address.

sjspielman committed:
ignore data-raw when checking deps

sjspielman committed:
Updates: add some #nolint, require profile, and use tempdir instead of tempfile

sjspielman committed:
profile -> aws_profile typo

sjspielman commented:
I updated to use `tempdir()` instead, as well as:

- I had to add `#nolint` for a couple long lines that we want the linter to skip; they should be as written
- I added an exclusion to the precommit hook `deps-in-desc`, which confirms that any `packages::` detected are in `DESCRIPTION`. I excluded the `data-raw/` directory since those scripts aren't part of the package API.

allyhawkins reviewed (approved):
LGTM
